experimental/dds/tree2/src/test/shared-tree/fuzz/targetedFuzz.spec.ts

```diff
@@ -270,6 +278,79 @@ describe("Fuzz - Targeted", () => {
 			emitter,
 		});
 	});
+
+	describe("out of order undo matches the initial state", () => {
+		const generatorFactory = (): AsyncGenerator<TreeOperation, UndoRedoFuzzTestState> =>
+			takeAsync(opsPerRun, makeOpGenerator(undoRedoWeights));
+
+		const model: DDSFuzzModel<
+			SharedTreeTestFactory,
+			Operation,
+			DDSFuzzTestState<SharedTreeTestFactory>
+		> = {
+			workloadName: "SharedTree",
+			factory: new SharedTreeTestFactory(onCreate),
+			generatorFactory,
+			reducer: fuzzReducer,
+			validateConsistency: () => {},
+		};
+		const emitter = new TypedEventEmitter<DDSFuzzHarnessEvents>();
+		emitter.on("testStart", (initialState: UndoRedoFuzzTestState) => {
+			initialState.initialTreeState = toJsonableTree(initialState.clients[0].channel);
+			initialState.firstAnchors = [];
+			// creates an initial anchor for each tree
+			for (const client of initialState.clients) {
+				initialState.firstAnchors.push(getFirstAnchor(client.channel));
+			}
+		});
+		emitter.on("testEnd", (finalState: UndoRedoFuzzTestState) => {
+			const clients = finalState.clients;
+
+			/**
+			 * TODO: Currently this array is used to track that undo() is called "opsPerRun" number of times.
+			 * Once the undo stack exposed, remove this array and use the stack to keep track instead.
+			 */
+			const undoOpsOnClients = new Array(clients.length).fill(0);
```

Comment:
i think it would be clearer to construct an array like

`const undoOrderByClientIndex = [1, 1, 1, 2, 2, 2, 3, 3, 3]` and then call `finalState.random.shuffle` on it.

Reply:
`Array.from({ length: opsPerRun * clients.length }, (_, index) => Math.floor(index / opsPerRun))` should do I believe

Reply:
updated to use an array with random client indices with length `opsPerRun * clients.length`